src/sentry/search/django/backend.py

```diff
@@ -250,6 +257,80 @@ def query(self, project, tags=None, environment=None, sort_by='date', limit=100,
         else:
             retention_window_start = None
 
+        # TODO: This could also do something janky like switching based on the current
+        #       tagstore backend, or a global option, or a number of other things. Going
+        #       by the project seemed the best for now.
+        use_snuba = bool(project.get_option('sentry:snuba', False))
+        if use_snuba and (tags or
+                          environment is not None or
+                          any(key in parameters for key in ('date_from', 'date_to'))):
+
+            # if we're in this branch then Snuba can help us filter down the eligible group ids
+            filters = {
+                'project_id': [project.id],
+            }
+
+            if environment is not None:
+                filters['environment'] = [environment.id]
+
+            conditions = []
+            for tag, val in six.iteritems(tags):
+                col = 'tags[{}]'.format(tag)
+                if val == ANY:
+                    conditions.append((col, 'IS NOT NULL', None))
+                else:
+                    conditions.append((col, '=', val))
+
+            now = timezone.now()
```

Comment:
Super pedantic comment, but we technically accept events with timestamps in the future (limited by some small window) so this technically doesn't return a comprehensive result set of all stored issues.

Reply:
Ahhh, and the old style just leaves out dates when they aren't required. Interesting.

Snuba requires dates, which I hope we keep. So maybe we just spread the default `now` by adding the window to it...?

Reply:
Yeah, that was sort of what I had in mind (adding time to `now` to cover the gap.)